torch/fx/experimental/proxy_tensor.py

```diff
@@ -409,7 +409,11 @@ def extract_val(val: _ExtractValType) -> _ExtractValType:
             # approach would be to maintain a per-trace FakeTensorMode and
             # from_real_tensor to create fake values (don't forget to
             # snapshot_fake)
-            fake_tensor_mode = FakeTensorMode(allow_fallback_kernels=True)
+            from torch._guards import detect_fake_mode
+
+            fake_tensor_mode = detect_fake_mode(val)
```

Comment:
i am kinda surprised how aot_export_module handles this case? aot_export_module internally calls make_fx, which I feel it should run into same issue. Can you try the same repro with torch.export._trace._export(pre_dispatch=False)?

Reply:
> pre_dispatch=False

@tugsbayasgalan  `torch.export._trace._export(pre_dispatch=False)?` works fine. The main reason is that in this case, when we get to make_fx, the args are already FakeTensors, but in my test case and the issue linked, when we get to make_fx, some args are Tensors, not FakeTensors.

Reply:
and they are converted to FakeTensors here:

https://github.com/pytorch/pytorch/blob/22580f160e9ff6f5a54bc5abd03ba3eb75519e10/torch/_functorch/aot_autograd.py#L1594

Reply:
I guess one alternative way we could fix this is we also convert all inputs to FakeTensors in `_export_to_aten_ir_make_fx` ?

Reply:
Yeah I think we should mimic aot_export_module as much as possible, so fakifying everything before passing them into make_fx makes sense to me.

Reply:
@tugsbayasgalan sure, I can make a PR for that. I think we should still also land this PR as well. WDTY?

Reply:
Sure! Could you leave a TODO with a task here tho?

Reply:
@tugsbayasgalan Some update on this, we actually tried to convert the input to FakeTensors here: 

https://github.com/pytorch/pytorch/blob/22580f160e9ff6f5a54bc5abd03ba3eb75519e10/torch/export/_trace.py#L228

The problem is that when the kwarg is `transformers.tokenization_utils_base.BatchEncoding`, the conversion failed. I'm looking into more on how to fix it.